src/hyp3_isce2/insar_stripmap.py

```diff
@@ -24,31 +32,84 @@
     os.environ['PATH'] = str(ISCE_APPLICATIONS) + os.pathsep + os.environ['PATH']
 
 
-def insar_stripmap(reference_scene: str, secondary_scene: str) -> Path:
+# TODO update docstring description
+def insar_stripmap(user: str, password: str, reference_scene: str, secondary_scene: str) -> Path:
     """Create an interferogram
 
     This is a placeholder function. It will be replaced with your actual scientific workflow.
 
     Args:
+        user: Earthdata username
+        password: Earthdata password
         reference_scene: Reference scene name
         secondary_scene: Secondary scene name
 
     Returns:
         Path to the output files
     """
+    session = asf_search.ASFSession().auth_with_creds(user, password)
+
+    reference_product, secondary_product = asf_search.search(
+        granule_list=[reference_scene, secondary_scene],
+        processingLevel=asf_search.L1_0,
+    )
+    assert reference_product.properties['sceneName'] == reference_scene
+    assert secondary_product.properties['sceneName'] == secondary_scene
+    products = (reference_product, secondary_product)
 
-    raise NotImplementedError('This is a placeholder function. Replace it with your actual scientific workflow.')
+    polygons = [Polygon(product.geometry['coordinates'][0]) for product in products]
+    insar_roi = polygons[0].intersection(polygons[1]).bounds
 
-    product_file = Path("product_file_name.zip")
-    return product_file
+    dem_path = download_dem_for_isce2(insar_roi, dem_name='glo_30', dem_dir=Path('dem'), buffer=0)
+
+    urls = [product.properties['url'] for product in products]
+    asf_search.download_urls(urls=urls, path=os.getcwd(), session=session, processes=2)
+
+    zip_paths = [product.properties['fileName'] for product in products]
+    for zip_path in zip_paths:
+        with zipfile.ZipFile(zip_path, 'r') as zip_file:
+            zip_file.extractall()
+        os.remove(zip_path)
+
+    reference_image = get_product_file(reference_product, 'IMG-')
+    reference_leader = get_product_file(reference_product, 'LED-')
+
+    secondary_image = get_product_file(secondary_product, 'IMG-')
+    secondary_leader = get_product_file(secondary_product, 'LED-')
+
+    config = stripmapapp.StripmapappConfig(
+        reference_image=reference_image,
+        reference_leader=reference_leader,
+        secondary_image=secondary_image,
+        secondary_leader=secondary_leader,
+        roi=insar_roi,
+        dem_filename=str(dem_path),
+    )
+    config_path = config.write_template('stripmapApp.xml')
+
+    stripmapapp.run_stripmapapp(start='startup', end='geocode', config_xml=config_path)
+
+    # TODO is this still needed?
+    #stripmapapp.run_stripmapapp(start='rubber_sheet_range', end='geocode', config_xml=config_path)
+
+    return Path('interferogram')
```

Comment:
I'm curious if we really want to use the `interferogram/` directory as our product directory (which will get zipped and uploaded to S3). I see that this directory contains the following files (including `highBand/` and `lowBand/` directories which contain even more files):

```
filt_topophase.flat
filt_topophase.flat.geo
filt_topophase.flat.geo.vrt
filt_topophase.flat.geo.xml
filt_topophase.flat.vrt
filt_topophase.flat.xml
filt_topophase.unw
filt_topophase.unw.conncomp
filt_topophase.unw.conncomp.geo
filt_topophase.unw.conncomp.geo.vrt
filt_topophase.unw.conncomp.geo.xml
filt_topophase.unw.conncomp.vrt
filt_topophase.unw.conncomp.xml
filt_topophase.unw.geo
filt_topophase.unw.geo.vrt
filt_topophase.unw.geo.xml
filt_topophase.unw.vrt
filt_topophase.unw.xml
highBand/
lowBand/
phsig.cor
phsig.cor.geo
phsig.cor.geo.vrt
phsig.cor.geo.xml
phsig.cor.vrt
phsig.cor.xml
topophase.amp
topophase.amp.vrt
topophase.amp.xml
topophase.cor
topophase.cor.full
topophase.cor.full.vrt
topophase.cor.full.xml
topophase.cor.geo
topophase.cor.geo.vrt
topophase.cor.geo.xml
topophase.cor.vrt
topophase.cor.xml
topophase.flat
topophase.flat.geo
topophase.flat.geo.vrt
topophase.flat.geo.xml
topophase.flat.vrt
topophase.flat.xml
```

Do we need to include all of these files in our product zipfile? If not, which files would we like to include?

Reply:
The highBand and lowBand folders are used for the ionospheric corrections but they are intermediate products so I think it's ok to remove them.  But maybe we should include some additional files. I'll check and give you a list of the products that should go to the bucket. Thanks!!

Reply:
Hi @jtherrmann I checked and I think we can discard the highBand and lowBand folders and keep the rest in the interferogram folder. Also, I think keeping the 'geometry' and 'ionosphere' folders would be helpful. The idea is to keep the input files for MintPy

Reply:
Hi @mfangaritav, sounds good. Would it be more convenient to keep the file structure as it is right now, or do we want to move all of those files into a zipfile? If you're just planning to run the workflow locally for now, then the first option might be more convenient, but if you want to run this workflow in the HyP3 AVO deployment, we'll want to move (or copy) those files into a separate zipfile.

Reply:
@mfangaritav I opened https://github.com/mfangaritav/hyp3-isce2/pull/3 to show what it might look like to move the important files into the output zip. Alternatively, we could copy (rather than move) the files in order to preserve the original file structure, if that would be more convenient for local testing. Or, if this workflow does not need to run in the cloud at all, then we could skip creating the output zip entirely.

Reply:
Thanks @jtherrmann for including the geometry folder. Yes, I think we should have the zip file as an output, because the idea is to run it in the cloud.